Tools/Cli-LoRa-Device-Provisioning/Helpers/IoTDeviceHelper.cs

```diff
@@ -904,34 +907,63 @@ public bool VerifyDevice(AddOptions opts, string fCntUpStartReported, string fCn
         public bool VerifyConcentrator(AddOptions opts)
         {
             var isValid = true;
-            if (string.IsNullOrEmpty(opts.StationEui))
+            TrackErrorIf(string.IsNullOrEmpty(opts.StationEui), "'Concentrator' device type has been specified but StationEui option is missing.");
+            TrackErrorIf(string.IsNullOrEmpty(opts.Region), "'Concentrator' device type has been specified but Region option is missing.");
+            TrackErrorIf(opts.Region is { } region && !File.Exists(Path.Combine(DefaultRouterConfigFolder, $"{region.ToUpperInvariant()}.json")),
+                         $"'Concentrator' device type has been specified with Region '{opts.Region.ToUpperInvariant()}' but no default router config file was found.");
+            if (opts.NoCups)
             {
-                StatusConsole.WriteLogLine(MessageType.Error, "'Concentrator' device type has been specified but StationEui option is missing.");
-                isValid = false;
+                TrackErrorIf(opts.TcUri is not null, "TC URI must not be defined if --no-cups is set.");
+                TrackErrorIf(opts.CupsUri is not null, "CUPS URI must not be defined if --no-cups is set.");
+                TrackErrorIf(!string.IsNullOrEmpty(opts.CertificateBundleLocation), "Certificate bundle location must not be defined if --no-cups is set.");
             }
-            if (string.IsNullOrEmpty(opts.Region))
+            else
             {
-                StatusConsole.WriteLogLine(MessageType.Error, "'Concentrator' device type has been specified but Region option is missing.");
-                isValid = false;
+                TrackErrorIf(!File.Exists(opts.CertificateBundleLocation), "CUPS is enabled but no certificate bundle was found at the specified location.");
+                TrackErrorIf(!opts.ClientCertificateThumbprints.Any(), "CUPS is enabled but no client certificate thumbprints were specified.");
+                TrackErrorIf(opts.TcUri is null, "CUPS is enabled but TC URI is not defined.");
+                TrackErrorIf(opts.TcUri is { } tcUri && !tcUri.Scheme.Equals("wss", StringComparison.OrdinalIgnoreCase), "CUPS is enabled but TC URI is not in wss:// protocol.");
+                TrackErrorIf(opts.CupsUri is null, "CUPS is enabled but CUPS URI is not defined");
+                TrackErrorIf(opts.CupsUri is { } cupsUri && !cupsUri.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase), "CUPS is enabled but CUPS URI is not in https:// protocol.");
```

Comment:
What do you think about verifying that tcUri and cupsUri are respectively pointing to :5001 and :443?